File: Projects/Project_Cinema Website/frontend/src/services/UserService.js
import axios from "axios";

const baseURL = 'http://localhost:8080/api/users'; // Replace with your actual API endpoint

class UserService {

    // GET
    static async getUserById(userId) {
        const response = await axios.get(`${baseURL}/getUser/${userId}`);
        return response.data;
    }

    // Authentication (pass username and password)
    static async getUserByCredentials(username, password) {
        const response = await axios.get(`${baseURL}/getUserByCredentials/${username}/${password}`);
        return response.data;
    }

    static async getAllUsers() {
        const response = await axios.get(`${baseURL}/getAllUsers`);
        return response.data;
    }

    // POST
    static async addUser(newUser) {
        const response = await axios.post(`${baseURL}/addUser`, newUser);
        return response.data;
    }

    // PUT (Update)
    static async updateUserById(userId, updatedUser) {
        const response = await axios.put(`${baseURL}/updateUser/${userId}`, updatedUser);
        return response.data;
    }

    static async updateUserStatus(userId, updatedStatus) {
        const response = await axios.put(`${baseURL}/updateUserStatus/${userId}`, updatedStatus, {
            headers: {
                'Content-Type': 'text/plain'
            }
        });
        return response.data;
    }

    // DELETE
    static async deleteUserById(userId) {
        const response = await axios.delete(`${baseURL}/deleteUser/${userId}`);
        return response.data; // Assuming backend returns status 200 for success
    }


}

export default UserService;

File: Projects/Project_Cinema Website/backend/src/main/java/com/cinema_website/backend/service/UserService.java
package com.cinema_website.backend.service;

import com.cinema_website.backend.dto.UserDTO;
import com.cinema_website.backend.mapper.UserMapper;
import com.cinema_website.backend.repository.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@SuppressWarnings("all")
@Service
public class UserService {

    private final UserRepository userRepository;
    private final UserMapper userMapper;

    @Autowired
    public UserService(UserRepository userRepository, UserMapper userMapper) {
        this.userRepository = userRepository;
        this.userMapper = userMapper;
    }

    // GET
    public UserDTO getUser(int userId) {
        return userRepository.getUserById(userId);
    }

    public UserDTO getUserByCredentials(String username, String password) {
        return userRepository.getUserByCredentials(username, password);
    }

    public UserDTO getUserByUsername(String username) {
        return userRepository.getUserByUsername(username);
    }

    public UserDTO getUserByEmail(String email) {
        return userRepository.getUserByEmail(email);
    }

    public List<UserDTO> getAllUsers() {
        return userRepository.getAllUsers();
    }

    // INSERT
    public boolean addUser(UserDTO user) {
        return userRepository.addUser(user);
    }

    // UPDATE
    public boolean updateUser(int userId, UserDTO updatedUser) {
        return userRepository.updateUserById(userId, updatedUser);
    }

    public boolean updateUserStatus(int userId, String updatedStatus) {
        return userRepository.updateUserStatus(userId, updatedStatus);
    }



    // DELETE
    public boolean deleteUser(int userId) {
        return userRepository.deleteUserById(userId);
    }

    // Update last login timestamp for authentication purposes
    public boolean updateLastLogin(int userId) {
        return userRepository.updateLastLogin(userId);
    }
}
